libsmtutil/SMTPortfolio.cpp

```diff
@@ -155,3 +158,12 @@ bool SMTPortfolio::solverAnswered(CheckResult result)
 {
 	return result == CheckResult::SATISFIABLE || result == CheckResult::UNSATISFIABLE;
 }
+
+string SMTPortfolio::dumpQuery(vector<Expression> const& _expressionsToEvaluate)
+{
+	// This code assumes that the constructor guarantees that
+	// SmtLib2Interface is in position 0, if enabled.
+	if (auto smtlib2 = dynamic_cast<SMTLib2Interface*>(m_solvers.front().get()))
```

Comment:
I think we should assert this here and have the caller guarantee that it is enabled. I don't think we should return "" here.